extensions/notebook/src/protocol/notebookUriHandler.ts

```diff
@@ -33,15 +32,25 @@ export class NotebookUriHandler implements vscode.UriHandler {
 				vscode.window.showErrorMessage(localize('notebook.unsupportedAction', "Action {0} is not supported for this handler", uri.path));
 		}
 	}
-
+	/**
+	 * Our Azuredatastudio URIs follow the standard URI format of
+	 * azuredatastudio://microsoft.notebook/open?url=...
+	 */
 	private open(uri: vscode.Uri): Promise<void> {
-		const data = querystring.parse(uri.query);
+		let data: string;
```

Comment:
Would love to see some more tests added covering the new scenarios you're supporting

Reply:
Yes, I can do that. I would like to work on tests on Testing Monday moreover for these specific scenarios and begin working on getting this PR in so that I can work with Stuart on his Executable TSGs.